elastalert/elastalert.py

```diff
@@ -784,7 +784,10 @@ def alert(self, matches, rule, alert_time=None):
             try:
                 alert.alert(matches)
             except EAException as e:
-                self.handle_error('Error while running alert %s: %s' % (alert.get_info()['type'], e), {'rule': rule['name']})
+                self.handle_error('ElastAlert Error while running alert %s: %s' % (alert.get_info()['type'], e), {'rule': rule['name']})
+                alert_exception = str(e)
+            except Exception as e:
+                self.handle_error('Unexpected Error while running alert %s: %s' % (alert.get_info()['type'], e), {'rule': rule['name']})
```

Comment:
I don't think that is necessarily what we want though. This part of the code base has no idea what the alerters beneath it are doing, so it doesn't really make sense to catch a specific error one of them might throw.